src/modules/BootSupportCheck.rb

```diff
@@ -79,11 +79,11 @@ def correct_loader_type(type)
       return true if type == "none"
 
       # grub2 is sooo cool...
-      return true if type == "grub2" && !Arch.aarch64 && !Arch.arm
+      return true if type == "grub2" && !Arch.aarch64 && !Arch.arm && !Arch.riscv64
 
       return true if (Arch.i386 || Arch.x86_64) && type == "grub2-efi" && efi?
 
-      return true if type == "grub2-efi" && (Arch.aarch64 || Arch.arm)
+      return true if type == "grub2-efi" && (Arch.aarch64 || Arch.arm || Arch.riscv64)
```

Comment:
As I see duplicities here, I would say that it makes sense to define method efi_only? in SystemInfo class and use it from there as I see it is used in multiple places.

Reply:
good point, changed.